server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -410,10 +410,11 @@ export class AgenticChatController implements ChatHandlers {
                 // depends on it
                 session.conversationId = uuid()
             }
-
+            const chatResultStream = this.#getChatResultStream(params.partialResultToken)
             token.onCancellationRequested(async () => {
                 this.#log('cancellation requested')
                 await this.#showUndoAllIfRequired(chatResultStream, session)
+                await chatResultStream.updateOngoingProgressResult('Canceled')
                 await this.#getChatResultStream(params.partialResultToken).writeResultBlock({
```

Comment:
nit: we can reuse the chatResultStream variable now?

Reply:
the this. inside the arrow function does not bind to the same context. 

`chatResultStream` and `this.#getChatResultStream(params.partialResultToken)` is not the same result stream. I don't have context why this was implemented in the first place, but I suggest we use the same `chatResultStream`

Reply:
okay, thanks for the explanation, but this is very confusing, we should fix it